kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -527,7 +545,32 @@ private void commitFile(String encodedPartition) {
       writers.remove(encodedPartition);
       log.debug("Removed writer for '{}'", encodedPartition);
     }
+  }
 
+  private void tagFile(String encodedPartition, String s3ObjectPath) {
+    if (tagging) {
+      if (!startOffsets.containsKey(encodedPartition)) {
+        log.warn("Tried to tag file with missing starting offset partition: {}. Ignoring",
+                encodedPartition);
+      } else {
+        log.debug("Object to tag is: {}", s3ObjectPath);
+        Map<String, String> tags = new HashMap<>();
+        tags.put("startOffset", Long.toString(startOffsets.get(encodedPartition)));
+        tags.put("endOffset", Long.toString(endOffsets.get(encodedPartition)));
+        tags.put("recordCount", Long.toString(recordCounts.get(encodedPartition)));
+
+        try {
+          storage.addTags(s3ObjectPath, tags);
+          log.info("Tagged file {} with record count {}", s3ObjectPath, recordCount);
+        } catch (SdkClientException e) {
+          log.trace("Unable to tag file {} with record count.", s3ObjectPath);
+        } catch (Exception e) {
+          log.trace("Unrecoverable exception while attempting to tag s3 object.");
+        }
+      }
+    }
     startOffsets.remove(encodedPartition);
```

Comment:
Keeping the responsibilities separate makes sense. However, placing the call back to the initial position in commitFile() would remove the values before they are used for tagging, since commitFile() is called before tagging is called.

Went ahead and moved it to commitFiles() instead, after both commitFile() call and the optional tagFile() call has been complete.

Reply:
Yeah, that's what I meant. Thanks!